app/ui/src/app/integration/import-export/import/integration-import.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+import { ApiHttpService } from '@syndesis/ui/platform';
+import { IntegrationImportData, IntegrationImportRequest } from './integration-import.models';
+
+@Injectable()
+export class IntegrationImportService {
+  constructor(private apiHttpService: ApiHttpService) { }
+
+  validateImportInfo(importRequest: IntegrationImportRequest): Observable<IntegrationImportData> {
+    const apiHttpService = this.apiHttpService.setEndpointUrl('validateImportInfo');
+    const { specificationFile, integrationTemplateId } = importRequest;
+    if (specificationFile) {
+      return apiHttpService.upload<IntegrationImportData>({
+        specification: specificationFile
+      }, {
+        connectorSettings: { integrationTemplateId }
+      });
+    } else {
+      return apiHttpService.post<IntegrationImportData>(importRequest);
+    }
+  }
+
+  importIntegration(importRequest: IntegrationImportRequest): Observable<any> {
+    const apiHttpService = this.apiHttpService.setEndpointUrl('submitImport');
+
+    const [importIntegration, icon, specification] = [
+      importRequest,
+      importRequest.iconFile,
+      importRequest.specificationFile
+    ];
```

Comment:
Beware with copy+pasted code, since that's the easiest way to introduce technical debt into a project: `iconfile` or `specificationFile` do not exist in the `IntegrationImportRequest` interface.

Reply:
I was trying to use the ApiConnector stuff as an example for following the Redux architecture, but ended up copying (and subsequently removing) a lot of the code. I think it would have been best to apply what I'm actively learning outside of a feature requirement.